service/src/main/java/bio/terra/workspace/service/resource/controlled/flight/clone/azure/container/CopyAzureStorageContainerDefinitionStep.java

```diff
@@ -0,0 +1,155 @@
+        buildDestinationControlledAzureContainer(
+            sourceContainer,
+            destStorageAccountId,
+            destinationWorkspaceId,
+            destinationResourceId,
+            destinationResourceName,
+            description,
+            destinationContainerName);
+    ApiAzureStorageContainerCreationParameters destinationCreationParameters =
+        new ApiAzureStorageContainerCreationParameters()
+            .storageContainerName(destinationContainerName)
+            .storageAccountId(destStorageAccountId);
+    ControlledResourceIamRole iamRole =
+        IamRoleUtils.getIamRoleForAccessScope(sourceContainer.getAccessScope());
+
+    // create the container
+    ControlledAzureStorageContainerResource clonedContainer =
+        controlledResourceService
+            .createControlledResourceSync(
+                destinationContainerResource, iamRole, userRequest, destinationCreationParameters)
+            .castByEnum(WsmResourceType.CONTROLLED_AZURE_STORAGE_CONTAINER);
+
+    // save the container definition for downstream steps
+    workingMap.put(
+        WorkspaceFlightMapKeys.ControlledResourceKeys.CLONED_RESOURCE_DEFINITION, clonedContainer);
+
+    var apiCreatedContainer =
+        new ApiCreatedControlledAzureStorageContainer()
+            .azureStorageContainer(clonedContainer.toApiResource())
+            .resourceId(destinationContainerResource.getResourceId());
+
+    var apiContainerResult =
+        new ApiClonedControlledAzureStorageContainer()
+            .effectiveCloningInstructions(resolvedCloningInstructions.toApiModel())
+            .storageContainer(apiCreatedContainer)
+            .sourceWorkspaceId(sourceContainer.getWorkspaceId())
+            .sourceResourceId(sourceContainer.getResourceId());
+
+    if (resolvedCloningInstructions.equals(CloningInstructions.COPY_DEFINITION)) {
+      FlightUtils.setResponse(flightContext, apiContainerResult, HttpStatus.OK);
+    }
+
+    return StepResult.getStepResultSuccess();
+  }
+
+  private void validateInputs(FlightMap inputParameters, FlightMap workingMap) {
+    FlightUtils.validateRequiredEntries(
+        inputParameters,
+        WorkspaceFlightMapKeys.ControlledResourceKeys.DESTINATION_WORKSPACE_ID,
+        WorkspaceFlightMapKeys.ControlledResourceKeys.DESTINATION_RESOURCE_ID,
+        WorkspaceFlightMapKeys.ControlledResourceKeys.DESTINATION_CONTAINER_NAME);
+    FlightUtils.validateRequiredEntries(
+        workingMap,
+        WorkspaceFlightMapKeys.ControlledResourceKeys.DESTINATION_STORAGE_ACCOUNT_RESOURCE_ID);
+  }
+
+  @Override
+  public StepResult undoStep(FlightContext context) throws InterruptedException {
+    var clonedContainer =
+        context
+            .getWorkingMap()
+            .get(
+                WorkspaceFlightMapKeys.ControlledResourceKeys.CLONED_RESOURCE_DEFINITION,
+                ControlledAzureStorageContainerResource.class);
+    if (clonedContainer != null) {
```

Comment:
Because you set the working map after the create, you will miss deleting this if the pod is stopped at line 106.
Ideally, you can save state in the map and then try the create.

Also, I think the undo needs to handle the not-found case, because it can be retried/rerun so the object might already be gone. That should not be an error. (Maybe double check that the create code does that right as well?)